samples/matter/common/src/Kconfig

```diff
@@ -40,18 +40,47 @@ config NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE
 	  Allow device to perform factory reset if the operational key for Fabric has not been migrated
 	  properly to PSA ITS storage.
 
-config NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
+menuconfig NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
 	bool "Enable diagnostics logs support"
 	help
 	  Provides support for diagnostics logs. Diagnostics logs allow the Matter controller to read end-user,
 	  network and crash logs from the Matter device.
 	  Diagnostic logs are usable only if the DiagnosticCluster is available and active on the device's root endpoint.
 	  Otherwise the compilation error occurs.
 
+if NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
+
 config NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_MAX_SIMULTANEOUS_SESSIONS
 	int "Define maximum simultaneous sessions"
 	default 20
 	help
 	  The maximum simultaneous sessions, which is the capability for sending diagnostic logs
 	  to Matter controller using multiple sessions at the same time for different diagnostic
 	  intents. When a session finishes, it releases its slot for a new one.
+
+config NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_CRASH_LOGS
+	bool "Enable crash logs support"
+	select RETAINED_MEM
+	select RETENTION
+	select RETENTION_MUTEX_FORCE_DISABLE
+	select RETAINED_MEM_MUTEX_FORCE_DISABLE
+	help
+	  Enables support for storing crash logs when the crash occurs.
+
+config NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_REMOVE_CRASH_AFTER_READ
```

Comment:
It's not a comment for your change or request, but just something came to my mind and I would like to share it. What is going to happen in the multi-admin scenario? By default only a single controller will be able to read the crash log. We can disable removing crash logs, but what will be the impact on the user? Who will be responsible for removing them? Or are they going to be replaced on the next crash?

Reply:
I added this config because I had the same considerations. It is hard to define who will be responsible for removing them because it is not defined in the spec, but there is only the note that the device should send the last log data available. and in my opinion, the final decision should be made by a product maker. And yes, the crash logs will be replaced on the next crash. So only the last one is available to be read out.